Godwoken syndrome will now stop functioning when RNG gets stuck instead of spamming.

interestingusernam3 opened the pull request:
<!-- Write **BELOW** The Headers and **ABOVE** The comments else it may not be viewable. -->
<!-- You can view Contributing.MD for a detailed description of the pull request process. -->

## About The Pull Request
Fixes #55392.
<!-- Describe The Pull Request. Please be sure every change is documented or this can delay review and even discourage maintainers from merging your PR! -->

## Why It's Good For The Game
Bugfix
<!-- Please add a short description of why you think these changes would benefit the game. If you can't justify it in words, it might not be worth adding. -->

## Changelog
:cl:
fix: Eldritch gods no longer speak fast enough to trigger spam filters.
/:cl:

<!-- Both :cl:'s are required for the changelog to work! You can put your name to the right of the first :cl: if you want to overwrite your GitHub username as author ingame. -->
<!-- You can use multiple of the same prefix (they're only used for the icon ingame) and delete the unneeded ones. Despite some of the tags, changelogs should generally represent how a player might be affected by the changes rather than a summary of the PR's contents. -->

Commit:
Slows down neutral VoG messages 4 times

Should no longer near spam filter, hopefully.

XDTM commented:
I swear byond's rng has a tendency to get stuck at times, the kind of situation described in the bug feels way too frequent when i play or test the game

interestingusernam3 commented:
Yeah this bug is spam of "YOU. STOP", so the RNG is clearly stuck on some number that  picks that phrase (out of that list, it's absolutely not special at all). This just makes it so the bug happens rarely (I hope). If there is some way to change BYOND RND from within this repo this should be closed as such a fix would clearly be prefferable.

XDTM commented:
not only that, but rolling a 4% chance with enough frequency to trigger the spam alert, on top of the chance of rolling the same phrase every time

ATH1909 commented:
Surely it'd be better to find the source of this problem and fix that, instead of just patching the symptoms?

Commit:
New fix

Replaces prob() with rand() and a comparison that does what the prob does when it works correctly. This should make the bug less common. I hope.

Commit:
Actually the previous new fix was stupid

Here's something that should actually work

interestingusernam3 commented:
>Surely it'd be better to find the source of this problem and fix that, instead of just papering over the symptoms?
I now made it call RNG twice and check if it's equal to two different values. The chance for success is still 4% but now it will never succeed when the  RNG gets stuck.

XDTM commented:
I don't know if using rand would change much, if the RNG engine does get stuck like i think, it'll keep generating the same result no matter the method. It's probably worth experimenting with, but this spot is a bit too niche to do so.

Honestly i'd probably reset it back to how it was and create a separate issue report for the RNG oddities, closing this one as part of it. If we do find a fix it will have to be used wherever we use probability, so it'll have to be a large refactor either way.

interestingusernam3 commented:
>I don't know if using rand would change much, if the RNG engine does get stuck like i think, it'll keep generating the same result no matter the method. It's probably worth experimenting with, but this spot is a bit too niche to do so.

That's why this fixes it. It first checks if the number is 2, then if it's 3. If it's stuck it will always fail. 

>Honestly i'd probably reset it back to how it was and create a separate issue report for the RNG oddities, closing this one as part of it. If we do find a fix it will have to be used wherever we use probability, so it'll have to be a large refactor either way.

This is a problem with BYOND and will therefoere probably never be fixed.

Jared-Fogle commented on code/datums/brain_damage/special.dm:
```diff
@@ -12,7 +12,7 @@
 /datum/brain_trauma/special/godwoken/on_life()
 	..()
-	if(prob(4))
+	if((rand(1,5)==2)&&rand(1,5==3)) //This check should fail if the RNG is stuck
```
5 == 3...? did you test this dude

Jared-Fogle commented:
what the fuck is this

Mothblocks reviewed (changes requested):
this doesn't fix the bug, it's just stupid. make random words of god not trigger the spam filter.
